dataflux_core/performance_tests/list_and_download.py

```diff
@@ -18,63 +18,120 @@
+        if config["list_timeout"] and listing_time > config["list_timeout"]:
             raise AssertionError(
-                f"Expected list operation to complete in under {list_timeout} seconds, but took {listing_time} seconds."
+                f"Expected list operation to complete in under {config['list_timeout']} seconds, but took {listing_time} seconds."
             )
-        download_params = download.DataFluxDownloadOptimizationParams(max_compose_bytes)
-        download_start_time = time.time()
-        download_result = download.dataflux_download(
-            project,
-            bucket,
-            list_result,
-            dataflux_download_optimization_params=download_params,
+        return list_result
+
+    def run_download(self, config, list_result):
+        download_params = download.DataFluxDownloadOptimizationParams(
+            config["max_compose_bytes"]
         )
+        download_start_time = time.time()
+        download_result = None
+        if config["parallelization"] and config["parallelization"] > 1:
+            download_result = download.dataflux_download_parallel(
+                config["project"],
+                config["bucket"],
+                list_result,
+                dataflux_download_optimization_params=download_params,
+                parallelization=config["parallelization"],
+            )
+        else:
+            download_result = download.dataflux_download(
+                config["project"],
+                config["bucket"],
+                list_result,
+                dataflux_download_optimization_params=download_params,
+            )
         download_end_time = time.time()
         downloading_time = download_end_time - download_start_time
         total_size = sum([len(x) for x in download_result])
-        if expected_file_size and total_size != expected_file_size:
+        if (
+            config["expected_total_size"]
+            and total_size != config["expected_total_size"]
+        ):
             raise AssertionError(
-                f"Expected {expected_file_size} bytes but got {total_size} bytes"
+                f"Expected {config['expected_total_size']} bytes but got {total_size} bytes"
             )
-        if download_timeout and downloading_time > download_timeout:
+        if config["download_timeout"] and downloading_time > config["download_timeout"]:
             raise AssertionError(
-                f"Expected download operation to complete in under {download_timeout} seconds, but took {downloading_time} seconds."
+                f"Expected download operation to complete in under {config['download_timeout']} seconds, but took {downloading_time} seconds."
             )
+
+    def test_list_and_download_one_shot(self):
+        config = self.get_config()
+        list_result = self.run_list(config)
+        self.run_download(config, list_result)
+
+    def test_list_and_download_segmented(self):
```

Comment:
nit: would be great to add a comment to explain the reason that we need to have segmented test to prevent OOM issues if we download a large dataset entirely at once.

Reply:
Good call